src/edu/csus/ecs/pc2/clics/API202306/EventFeedStreamer.java

```diff
@@ -0,0 +1,1061 @@
+import edu.csus.ecs.pc2.core.model.ClarificationEvent;
+import edu.csus.ecs.pc2.core.model.ClientType;
+import edu.csus.ecs.pc2.core.model.ContestInformationEvent;
+import edu.csus.ecs.pc2.core.model.ContestTimeEvent;
+import edu.csus.ecs.pc2.core.model.Group;
+import edu.csus.ecs.pc2.core.model.GroupEvent;
+import edu.csus.ecs.pc2.core.model.IAccountListener;
+import edu.csus.ecs.pc2.core.model.IClarificationListener;
+import edu.csus.ecs.pc2.core.model.IContestInformationListener;
+import edu.csus.ecs.pc2.core.model.IContestTimeListener;
+import edu.csus.ecs.pc2.core.model.IGroupListener;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.core.model.IJudgementListener;
+import edu.csus.ecs.pc2.core.model.ILanguageListener;
+import edu.csus.ecs.pc2.core.model.IProblemListener;
+import edu.csus.ecs.pc2.core.model.IRunListener;
+import edu.csus.ecs.pc2.core.model.Judgement;
+import edu.csus.ecs.pc2.core.model.JudgementEvent;
+import edu.csus.ecs.pc2.core.model.Language;
+import edu.csus.ecs.pc2.core.model.LanguageEvent;
+import edu.csus.ecs.pc2.core.model.Problem;
+import edu.csus.ecs.pc2.core.model.ProblemEvent;
+import edu.csus.ecs.pc2.core.model.Run;
+import edu.csus.ecs.pc2.core.model.RunEvent;
+import edu.csus.ecs.pc2.core.model.RunTestCase;
+import edu.csus.ecs.pc2.core.security.Permission;
+import edu.csus.ecs.pc2.core.util.IJSONTool;
+import edu.csus.ecs.pc2.services.web.IEventFeedStreamer;
+import edu.csus.ecs.pc2.ui.UIPlugin;
+
+/**
+ * Event Feed Server/Streamer.
+ *
+ * Constructs and sends event feed JSON to client via a stream.
+ *
+ * @author Douglas A. Lane, PC^2 Team, pc2@ecs.csus.edu
+ */
+public class EventFeedStreamer extends JSON202306Utilities implements Runnable, UIPlugin, IEventFeedStreamer {
+
+    /**
+     * Steps to provide a memento for the event feed. add a stream and filter per connection
+     */
+
+    private static final long serialVersionUID = 2076470194640278897L;
+
+    /**
+     * Number of ms be sending out keep alive new line.
+     */
+    private static final long KEEP_ALIVE_DELAY = 100 * Constants.MS_PER_SECOND;
+
+    /**
+     * Number of seconds between checks to send keep alive.
+     */
+    private static final int KEEP_ALIVE_QUERY_PERIOD_SECONDS = 25;
+
+    private Log log;
+
+    private IInternalContest contest;
+
+    // Listeners
+
+    private AccountListener accountListener = new AccountListener();
+
+    private RunListener runListener = new RunListener();
+
+    private ClarificationListener clarificationListener = new ClarificationListener();
+
+    private ProblemListener problemListener = new ProblemListener();
+
+    private LanguageListener languageListener = new LanguageListener();
+
+    private GroupListener groupListener = new GroupListener();
+
+    private JudgementListener judgementListener = new JudgementListener();
+
+    private ContestInformationListener contestInformationListener = new ContestInformationListener();
```

Comment:
I see that the `ContestInformationListener` handles generating events for type `CONTEST` and for type `STATE`.  I also see that there are listeners to handle events of type `ACCOUNTS` (teams), `SUBMISSIONS`  (the `RunListener`), `CLARIFICATIONS`, `PROBLEMS`, `LANGUAGES`, `GROUPS`, and `JUDGEMENTS`.  What's not clear to me is how events for `JUDGEMENT_TYPES` and `ORGANIZATIONS` are generated.

This may be just more of my lack of understanding of how the event feed works...

Reply:
See `JudgementListener` in `EventFeedStreamer` around line 720.   There is currently no way (that I am aware of) to add an organization to a contest that has been configured already.  (IE no `IOrganizationListener`).  Organizations are added during configuration only.  Organizations events are only generated once when the EF is started and that's it.  When organizations (institutions) are added the correct way and given their own model I suppose we'll add a listener for them then, but BFTF...